aiida/engine/processes/workchains/restart.py

```diff
@@ -57,13 +58,21 @@ class BaseRestartWorkChain(WorkChain):
     process will be run with those inputs.
 
     The `_process_class` attribute should be set to the `Process` class that should be run in the loop.
+    Finally, to define handlers that will be called during the `inspect_process` simply define a class method with the
+    signature `(self, node)` and decorate it with the `process_handler` decorator, for example::
+
+        @process_handler
+        def handle_problem(self, node):
```

Comment:
After the offline discussion yesterday I thought that the `self` argument is not needed anymore. Is it wrong?

I have no problem with either of the approaches, just got confused.

Reply:
It is still needed, but simply because it is an instance method. But now it should no longer come as a surprise since you have to directly define it within the class.